src/args.py

```diff
@@ -41,7 +54,18 @@
 #
 # Google "python keyword arguments".
 
-# YOUR CODE HERE
+
+def f4(*args, **kwargs):
```

Comment:
add a print() statement after every 'if' clause to see what's printing. I think you'll find that in the last call f4 call, f4(d), what's happening is not exactly what you thought (I may be wrong, but it looks like it from the way you wrote the code.)

Reply:
I took the instructions to be asking to set up the function so it could take keyword arguments, or positional arguments made up of dicts.

If I run the code, I get this output:
```
key: a, value: 12
key: b, value: 30
key: city, value: Berkeley
key: population, value: 121240
key: founded, value: March 23, 1868
key: monster, value: goblin
key: hp, value: 3
```
what are you seeing the intention as? to change how the argument is passed? something like `f4(monster=d['monster'], hp=d['hp')`?